pkg/apis/pipeline/v1alpha1/build_gcs_resource.go

```diff
@@ -0,0 +1,182 @@
+func NewBuildGCSResource(r *PipelineResource) (*BuildGCSResource, error) {
+	if r.Spec.Type != PipelineResourceTypeStorage {
+		return nil, fmt.Errorf("BuildGCSResource: Cannot create a BuildGCS resource from a %s Pipeline Resource", r.Spec.Type)
+	}
+	if r.Spec.SecretParams != nil {
+		return nil, fmt.Errorf("BuildGCSResource: %s cannot support artifacts on private bucket", r.Name)
+	}
+	var location, destDir string
+	var aType GCSArtifactType
+
+	for _, param := range r.Spec.Params {
+		switch {
+		case strings.EqualFold(param.Name, "Location"):
+			location = param.Value
+		case strings.EqualFold(param.Name, "DestinationDir"):
+			destDir = param.Value
+		case strings.EqualFold(param.Name, "ArtifactType"):
+			var err error
+			aType, err = getArtifactType(param.Value)
+			if err != nil {
+				return nil, fmt.Errorf("BuildGCSResource %s : %s", r.Name, err)
+			}
+		}
+	}
+	if location == "" {
+		return nil, fmt.Errorf("BuildGCSResource: Need Location to be specified in order to create BuildGCS resource %s", r.Name)
+	}
+	if aType == EmptyArtifactType {
+		return nil, fmt.Errorf("BuildGCSResource: Need ArtifactType to be specified in order to fetch BuildGCS resource %s", r.Name)
+	}
+	return &BuildGCSResource{
+		Name:           r.Name,
+		Type:           r.Spec.Type,
+		Location:       location,
+		DestinationDir: destDir,
+		ArtifactType:   aType,
+		Secrets:        r.Spec.SecretParams,
+	}, nil
+}
+
+// GetName returns the name of the resource
+func (s BuildGCSResource) GetName() string {
+	return s.Name
+}
+
+// GetType returns the type of the resource, in this case "storage"
+func (s BuildGCSResource) GetType() PipelineResourceType {
+	return PipelineResourceTypeStorage
+}
+
+// GetParams get params
+func (s *BuildGCSResource) GetParams() []Param { return []Param{} }
+
+// GetSecretParams returns the resource secret params
+func (s *BuildGCSResource) GetSecretParams() []SecretParam { return nil }
+
+// Replacements is used for template replacement on an GCSResource inside of a Taskrun.
+func (s *BuildGCSResource) Replacements() map[string]string {
+	return map[string]string{
+		"name":     s.Name,
+		"type":     string(s.Type),
+		"location": s.Location,
+	}
+}
+
+// SetDestinationDirectory sets the destination directory at runtime like where is the resource going to be copied to
+func (s *BuildGCSResource) SetDestinationDirectory(destDir string) { s.DestinationDir = destDir }
+
+// GetDownloadContainerSpec returns an array of container specs to download gcs storage object
+func (s *BuildGCSResource) GetDownloadContainerSpec() ([]corev1.Container, error) {
+	if s.DestinationDir == "" {
+		return nil, fmt.Errorf("BuildGCSResource: Expect Destination Directory param to be set %s", s.Name)
+	}
+	args := []string{"--type", string(s.ArtifactType), "--location", s.Location}
+	// dest_dir is the destination directory for GCS files to be copies"
+	if s.DestinationDir != "" {
```

Comment:
In line 136 code is already checking for `s.DestinationDir ` being empty string. Here  `if ` condition is always going to be true so code can be simplified to append `--dest_dir` args directly

Reply:
umm that is the `DownloadContainerSpec` flow.
Did you mean in L78 i am checking if DestinationDir is set? As per you suggestion below where you mentioned its a runtime config, i dropped that check on L78.
This check now  required. Does that make sense?